crates/context/config/src/client/near.rs

```diff
@@ -0,0 +1,287 @@
+use std::borrow::Cow;
+use std::collections::BTreeMap;
+use std::time;
+
+pub use near_crypto::SecretKey;
+use near_crypto::{InMemorySigner, Signer};
+use near_jsonrpc_client::{methods, JsonRpcClient};
+use near_jsonrpc_primitives::types::query::{QueryResponseKind, RpcQueryResponse};
+use near_jsonrpc_primitives::types::transactions::{RpcTransactionError, TransactionInfo};
+use near_primitives::action::{Action, FunctionCallAction};
+use near_primitives::transaction::{Transaction, TransactionV0};
+pub use near_primitives::types::AccountId;
+use near_primitives::types::{BlockReference, FunctionArgs};
+use near_primitives::views::{
+    AccessKeyPermissionView, AccessKeyView, CallResult, FinalExecutionStatus, QueryRequest,
+    TxExecutionStatus,
+};
+use thiserror::Error;
+use url::Url;
+
+use super::{Operation, Transport, TransportRequest};
+
+#[derive(Debug)]
+pub struct NetworkConfig {
+    pub rpc_url: Url,
+    pub account_id: AccountId,
+    pub access_key: SecretKey,
+}
+
+#[derive(Debug)]
+pub struct NearConfig<'a> {
+    pub networks: BTreeMap<Cow<'a, str>, NetworkConfig>,
+}
+
+#[derive(Debug)]
+struct Network {
+    client: JsonRpcClient,
+    account_id: AccountId,
+    secret_key: SecretKey,
+}
+
+#[derive(Debug)]
+pub struct NearTransport<'a> {
+    networks: BTreeMap<Cow<'a, str>, Network>,
+}
+
+impl<'a> NearTransport<'a> {
+    pub fn new(config: &NearConfig<'a>) -> Self {
+        let mut networks = BTreeMap::new();
+
+        for (network_id, network_config) in &config.networks {
+            let client = JsonRpcClient::connect(network_config.rpc_url.clone());
+
+            let _ignored = networks.insert(
+                network_id.clone(),
+                Network {
+                    client,
+                    account_id: network_config.account_id.clone(),
+                    secret_key: network_config.access_key.clone(),
+                },
+            );
+        }
+
+        Self { networks }
+    }
+}
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("unknown network: {0}")]
+    UnknownNetwork(String),
+    #[error("invalid response from RPC while {operation}")]
+    InvalidResponse { operation: String },
+    #[error("invalid contract ID: {0}")]
+    InvalidContractId(near_primitives::account::id::ParseAccountError),
+    #[error("failed while {operation}: {reason}")]
+    Custom {
+        operation: &'static str,
+        reason: String,
+    },
```

Comment:
Minor nit - it would be useful to keep variants in alphabetical order so that as they grow they are easier to find and manage 🙂

Reply:
That should be the job of the formatting tool. It will be available soon https://github.com/rust-lang/rustfmt/pull/6313